backend/authentication/auth_controller.js

```diff
@@ -187,67 +190,37 @@ exports.updateProfile = async (req, res) => {
       data: newProfileData,
     });
 
-    if (categories && categories.length > 0) {
-      await prisma.user.update({
+    if (interests && interests.length > 0) {
+      await prisma.interest.deleteMany({
         where: {
-          username: username,
-        },
-        data: {
-          categories: {
-            set: [],
-          },
+          userId: user.id,
         },
       });
-      for (category of categories) {
-        try {
-          await prisma.category.create({
-            data: {
-              category: category,
-              User: {
-                connect: {
-                  username: username,
-                },
-              },
-            },
-          });
-        } catch (error) {
-          res.status(500).json({
-            error: "Error updating categories",
-          });
-        }
-      }
+
+      const interestData = interests.map((interest) => ({
+        interest: interest,
+        userId: user.id,
+      }));
+      await prisma.interest.createMany({
+        data: interestData,
+      });
```

Comment:
should we still wrap the updates with a try catch?

Reply:
Sorry, I do not understand what you mean. Are suggesting I include try catch, like it was in the old version?

Reply:
yup, or it doesn't throw on failure?

Reply:
It does. I have updated it in this commit https://github.com/izuchukwumba/peered/pull/18/commits/27c7443539a37c234d407910f1468db115188954